### PR TITLE
Update browser title to use alternate title if it is available for SEO

### PR DESCRIPTION
**Pull request by Princesso**
Screenshot 2021-07-19 at 19 39 35

### CODE REVIEW

File: components/Article.tsx [hunk 1]
```diff
@@ -54,7 +55,7 @@ const Article: React.FunctionComponent<Props> = props => {
         )
     }
     return (
-        <PageLayout contentTitle={props.title} metaTags={metaTags} leftColumn={tocFragment}>
+        <PageLayout contentTitle={props.alternateTitle ? props.alternateTitle : props.title} metaTags={metaTags} leftColumn={tocFragment}>
```
Comment (marekweb) [inline components/Article.tsx:58]: Suggestion: would it make sense to simplify this with the `??` operator?

```suggestion
        <PageLayout contentTitle={props.alternateTitle ?? props.title} metaTags={metaTags} leftColumn={tocFragment}>
```

File: components/AuthorCardList.tsx [hunk 2]
```diff
@@ -16,6 +16,7 @@ const AuthorCardList: React.FunctionComponent<Props> = props => (
                 <AuthorCard
                     name={author.name}
                     bio={author.bio}
+                    id={author.id}
```
Comment (marekweb) [inline components/AuthorCardList.tsx:19]: Was this intended to be in the authors PR, instead? In any case, it seems fine to include it here, looks good.

**Review by marekweb — COMMENTED** [review]
Looks good!

My only question is about the best title format for the alternate title.

```
Regular Expressions, How To Search with Sourcegraph using Regular Expression Patterns, Sourcegraph
```

I don't know if we should use comma-separated keyword phrases, and if the alternate title should be longer than the regular title or if it should be shorter. I'll defer to @ltagliaferri on this question

**Comment by marekweb** [comment]
In the future we could think about whether the `alternateTitle` should keep the site suffix or not.

Currently the site suffix (`Sourcegraph Learn`) is added to the title tag. So a title (or alternate title) of `Lorem Ipsum` is transformed into `Lorem Ipsum - Sourcegraph Learn`.

https://github.com/sourcegraph/learn/blob/1f4a0f7dd7051bf75279d39af5fc8be507d8274f/components/Layout.tsx#L38-L44

When `alternateTitle` is used, should it omit the site suffix? That would give the post author full control over the title tag.

That's a suggestion outside of the scope of this PR.

File: components/Article.tsx (hunk 1 above)
Comment (Princesso) [reply components/Article.tsx:58]: Yeah, that makes sense. I will update this.

File: components/AuthorCardList.tsx (hunk 2 above)
Comment (Princesso) [reply components/AuthorCardList.tsx:19]: It was! The branch has been deleted after it was merged to main, so I thought I could just include it here.

**Comment by Princesso** [comment]
> In the future we could think about whether the `alternateTitle` should keep the site suffix or not.
> 
> Currently the site suffix (`Sourcegraph Learn`) is added to the title tag. So a title (or alternate title) of `Lorem Ipsum` is transformed into `Lorem Ipsum - Sourcegraph Learn`.
> 
> https://github.com/sourcegraph/learn/blob/1f4a0f7dd7051bf75279d39af5fc8be507d8274f/components/Layout.tsx#L38-L44
> 
> When `alternateTitle` is used, should it omit the site suffix? That would give the post author full control over the title tag.
> 
> That's a suggestion outside of the scope of this PR.

I think this is a discussion, we can have with the entire team.

File: posts/regular-expression-patterns.md [hunk 3]
```diff
@@ -5,6 +5,7 @@ tags: [tutorial, search, Sourcegraph]
 description: Learn how to use regular expression search patterns to search code on Sourcegraph.
 image: /headers/sourcegraph-learn-header-5.svg
 socialImage: /headers/sourcegraph-learn-header-5.png
+alternateTitle: Regular Expressions, How To Search with Sourcegraph using Regular Expression Patterns, Sourcegraph
```
Comment (ltagliaferri) [inline posts/regular-expression-patterns.md:8]: To optimize alternative/browser titles for SEO, we would use different keywords, as, for example "Using RegEx with Sourcegraph" that way "regex" in the browser title and "regular expressions" in the normal title would support SEO

**Comment by ltagliaferri** [comment]
Agree, we should omit the site suffix in alternative titles

**Pull request merged**